OpenMPI-related warnings

Hi everyone!

When I compile the AMReX-based code WarpX on my laptop, I get a lot of OpenMPI-related warnings, such as (see https://github.com/ECP-WarpX/WarpX/issues/1230):
```
/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi/ompi/mpi/cxx/op_inln.h:123:59: warning: cast between incompatible function types from ‘void (*)(const void*, void*, int, const MPI::Datatype&)’ to ‘void (*)(void*, void*, int*, ompi_datatype_t**)’ [-Wcast-function-type]
  123 |     ompi_op_set_cxx_callback(mpi_op, (MPI_User_function*) func);
      |                                                           ^~~~
In file included from /usr/lib/x86_64-linux-gnu/openmpi/include/openmpi/ompi/mpi/cxx/mpicxx.h:277,
                 from /usr/lib/x86_64-linux-gnu/openmpi/include/mpi.h:2868,
                 from ../amrex/Src/Base/AMReX_ccse-mpi.H:13,
                 from ../amrex/Src/Base/AMReX.H:8,
                 from ../amrex/Src/Base/AMReX_BLassert.H:5,
                 from ../amrex/Src/Base/AMReX_Arena.H:4,
                 from ../amrex/Src/Base/AMReX_GpuFuse.H:4,
                 from ../amrex/Src/Base/AMReX_Gpu.H:16,
                 from ./Source/Parser/GpuParser.H:12,
                 from ./Source/Parser/WarpXParserWrapper.H:11,
                 from ./Source/Particles/WarpXParticleContainer.H:13,
                 from ./Source/Particles/MultiParticleContainer.H:14,
                 from ./Source/Diagnostics/FlushFormats/FlushFormat.H:4,
                 from ./Source/Diagnostics/Diagnostics.H:4,
                 from ./Source/Diagnostics/Diagnostics.cpp:1:
/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi/ompi/mpi/cxx/op_inln.h: In member function ‘virtual void MPI::Op::Init(void (*)(const void*, void*, int, const MPI::Datatype&), bool)’:
/usr/lib/x86_64-linux-gnu/openmpi/include/openmpi/ompi/mpi/cxx/op_inln.h:121:46: warning: cast between incompatible function types from ‘void (*)(void*, void*, int*, ompi_datatype_t**, void (*)(void*, void*, int*, ompi_datatype_t**))’ to ‘void (*)(void*, void*, int*, ompi_datatype_t**)’ [-Wcast-function-type]
  121 |     (void)MPI_Op_create((MPI_User_function*) ompi_mpi_cxx_op_intercept,
```

Compiling with  `-DOMPI_SKIP_MPICXX` should effectively silence these warnings ( https://github.com/open-mpi/ompi/issues/5157) .  For this reason, I would like to ask if it could be possible to add something like:
```
ifeq ($(USE_MPI), TRUE)
  DEFINES += -DOMPI_SKIP_MPICXX
endif

```
to `GNUMake` in `AMReX`, in order to get rid of this noisy compilation warning.

--- a/Tools/GNUMake/sites/Make.unknown
+++ b/Tools/GNUMake/sites/Make.unknown
@@ -77,6 +77,8 @@ ifeq ($(USE_MPI),TRUE)
       LIBRARIES += $(filter -l%,$(mpicxx_link_flags))
 #    endif
 
+     DEFINES += -DOMPI_SKIP_MPICXX
+
   else ifneq ($(findstring Spectrum MPI, $(shell $(MPI_OTHER_COMP) -showme:version 2>&1)),)
 
     #

